xterm-color.el

```diff
@@ -310,6 +314,24 @@ inverse-color, frame, overline SGR state machine bits.")
        (push (list pos (text-properties-at pos string) (substring string pos)) result)
        (cl-return-from xterm-color--string-properties (nreverse result))))))
 
+(defun xterm-color--convert-text-properties-to-overlays (beg end)
+  "Delete face text properties between BEG and END, replacing with equivalent overlays."
```

Comment:
My current motivation for adding this function is for usage with Magit. When displaying a diff, Magit currently translates ANSI color sequences received from git into overlays (using the `ansi-color` library that is in Emacs). I am making it possible for Magit to use `xterm-color` instead of `ansi-color`, but I don't want to conflate that change with a change from overlays to text properties in the Magit diff buffer. Indeed, empirically, I see that if I set `xterm-color` to create text properties instead of overlays then the fontification in the Magit diff buffer  no longer works, so I suspect that Magit's choice of overlays may be deliberate. I think that other users of `xterm-color` will also need to be able to make this choice.